urle381b6cba7_IkarosKappler_ikrs_http_tgz-pJ8-ikrs_json_rpc_JSONRPCHandlerJ8/src/ikrs/json/JSONNumber.java

```
@@ -0,0 +1,201 @@
+     * respective Double.parseDouble(String). It expects a String containing
+     * a number and converts it into a Java Number object.
+     *
+     * If the passed string does not represent a number the method throws
+     * a NumberFormatException.
+     *
+     * @param str The string to be parsed.
+     * @return The parsed number value.
+     * @throws NullPointerException If the passed string is null.
+     * @throws JSONException If the passed string does not represent a number.
+     **/
+    public static Number parseNumber( String str )
+	throws NullPointerException,
+	       NumberFormatException {
+
+	if( str == null )
+	    throw new NullPointerException( "Cannot parse null to a number." );
+	
+	try {
+
+	    // Try to parse an integer first
+	    return Integer.parseInt( str );
+
+	} catch( NumberFormatException e ) {
+
+	    // Was not an integer. Try to parse floating point number then.
+	    return Double.parseDouble( str );
+
+	}
+	
+    }
+
+    /**
+     * This static method is the equivalent to Java's Integer.parseInt(String)
+     * respective Double.parseDouble(String). It expects a String containing
+     * a number and converts it into a JSONNumber object.
+     *
+     * If the passed string does not represent a number the method throws
+     * a JSONException (would be NumberFormatException in Java).
+     *
+     * @param str The string to be parsed.
+     * @return The parsed JSON number value.
+     * @throws NullPointerException If the passed string is null.
+     * @throws JSONException If the passed string does not represent a number.
+     **/
+    public static JSONNumber parseJSONNumber( String str )
+	throws NullPointerException,
+	       JSONException {
+
+	
+	try {
+	    // Try to convert string into a java value.
+	    Number number = JSONNumber.parseNumber( str );
+
+	    return new JSONNumber( number );
+	    
+	} catch( NumberFormatException e ) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)